packages/cozy-harvest-lib/src/helpers/accounts.js

```diff
@@ -90,11 +91,21 @@ export const mergeAuth = (account, authData) => ({
  * @return {object}           io.cozy.accounts attributes
  */
 export const updateTwoFaCode = (account, code) => ({
-  ...account,
-  // reset the state,
-  state: null,
+  // reset the state since the konnector is listening it
+  ...resetState(account),
```

Comment:
It's a bit surprising from the exterior that the `state` property (with no connection to 2fa, at least on the name) changes the state. I am sure it is surprising for me because I am not completely familiar with the business rules behind, but I think it would be more understandable like this :

```js
// This properties are special and reset the state
const shouldPropertyUpdateResetState = {
  twofa_code: true
}

export const setProperty = (account, name, property) => {
  const res = {...account, [property]: name }
  if (shouldPropertyUpdateResetState[property]) {
    res[state] = null
  }
  return res
}
```

It is still a bit surprising to have the update of a property change another but at least it is very explicit and you can have all the properties that do that at the same place. What do you think ?

Reply:
Harvest will listen to a state `TWO_NEEDED` to ask for the 2FA code and put it in the account, it's why setting the code should reset the state here.
For now this is the only one case and your suggestion is for me not more understandable, maybe when we will have more special properties with state reset but for now it's not the case.